docs/examples/nft-marketplace/nft-application.md

```diff
@@ -0,0 +1,710 @@
+                                <div className="image-preview">
+                                    <img src={URL.createObjectURL(image)} alt="nft" style={{width: 100, height: 100}}/>
+                                </div> 
+                            ): (
+                                <p>No image set for this NFT</p>
+                            )}
+                        </div>
+                        <Button type="submit" text="Create" className={styles.button}/>
+                    </form>
+                </div>
+            </>
+        )
+    }
+    ```
+
+In the next section, we will create the `Home` page where we will read and display the minted NFTs.
+
+### Home page
+
+In the `api.ts` file, we will add hooks for reading the contract state.
+
+1. First, let's add `useNFTState<T>`, where `T` is the type we expect to read (for example, `Token`). It will accept the function name and payload if required for the specified function:
+
+    ```typescript
+    import stateMetaWasm from 'assets/wasm/nft_state.meta.wasm'
+    import { useMetadata, useWasmMetadata } from './useMetadata'
+    import metaTxt from 'assets/meta/meta.txt'
+    import { useAccount, useReadWasmState, useSendMessage } from '@gear-js/react-hooks';
+    import { ADDRESS } from 'consts';
+
+    function useNFTMetadata() {
+        return useMetadata(metaTxt);
+    }
+
+    function useNFTState<T>(functionName: string, payload?: any) {
+        const { buffer } = useWasmMetadata(stateMetaWasm);
+        return useReadWasmState<T>(
+            ADDRESS.CONTRACT_ADDRESS,
+            buffer,
+            functionName,
+            payload
+        )
+    }
+    ```
+
+2. Let's read all the tokens our contract has. First, we'll create the type for a token in a separate folder called `types`:
+
+    ```shell
+    mkdir types
+    touch types/index.ts
+    ```
+
+    and add the `Token` description to the `index.ts` file:
+
+    ```typescript
+    import { HexString } from "@polkadot/util/types";
+
+    type Token = {
+        approvedAccountIds: HexString[];
+        description: string;
+        id: string;
+        media: string;
+        name: string;
+        ownerId: HexString;
+        reference: string;
+    };
+
+    export type { Token };
+    ```
+
+3. Then we can write the `useNFTs` hook:
+
+    ```typescript
+    ...
+    import { Token } from 'types';
+
+    ...
+    function useNFTs() {
+        const { state } = useNFTState<Token[]>("all_tokens", null);
+        return state;
+    }
+    ```
+
+4. Now let's start writing the `Home` page:
+
+    ```typescript
+
+    import { Loader } from 'components';
+    import { useNFTs } from 'hooks/api';
+    import styles from './Home.module.scss'
+
+    function Home() {
+
+    const nfts = useNFTs();
+    const isAnyNft = !!nfts?.length;
+
+    return (
+        <>
+        <header className={styles.header}>
+            <h2 className={styles.heading}>NFTs</h2>
+        </header>
+        {nfts ? (
```

Comment:
```suggestion
        {isNftStateRead ? (
```
While original code working as expected cuz state is undefined until it's read, I believe for better understanding and readability it's worth to look at read status